internal/pkg/cli/deploy/workload.go

```diff
@@ -342,8 +343,12 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 		return nil
 	}
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+	dockerCmdClient := dockerengine.New(exec.NewCmd())
+	if err := d.repository.Login(dockerCmdClient); err != nil {
```

Comment:
i wonder if `Login` should return the URI and set it in `buildArgs`? otherwise i think we'll make the call to ECR for every single container...

Reply:
`Login` will be called only once right for all the images. `Login` function is not inside any `loop` then how the login will be called for every container 🤔

Reply:
but we call `repository.URI()` every time we call `BuildAndPush()`! and `repository.URI()` makes an API call to ECR.